core/src/main/scala/chisel3/Annotation.scala

```diff
@@ -20,6 +20,14 @@ trait ChiselAnnotation {
   def toFirrtl: Annotation
 }
 
+/** Enhanced interface for Annotations in Chisel
+  *
+  *  Defines a conversion to corresponding FIRRTL Annotation(s)
+  */
+trait ChiselToFirrtlAnnotations {
+  def toFirrtlAnnotations: Seq[Annotation]
+}
```

Comment:
I'm not huge on this trait name, but I can't think of a better one. Maybe `FirrtlAnnotationGenerator` or `FirrtlAnnotationAdder` or `FirrtlAnnotationBuilder`. The method name would then possibly be better as `addFirrtlAnnotations` or something.

Reply:
@seldridge  don't have strong opinions on the name - I just went with something quick to reduce mental load!

Reply:
When we originally added the Annotation API, I actually argued for `() => firrtl.Annotation` as the type but was shot down in the Chisel dev meeting 🙃. That being said your slick trick for `.toRelativeTarget` won't work if we just use nullary functions so that might be a good enough reason to keep this new API as a trait. The coolest thing about your trick isn't the implicit class, it's that the `ChiselAnnotation` object can eagerly grab the current Module which is the biggest problem with `toRelativeTarget` (that grabbing the current module needs to be eager, but the .toRelativeTarget call needs to be lazy).

Reply:
ok - I admit to not going over/understanding  @seldridge  trick yet for `toRelativeTarget` - and will just agree with @jackkoenig 's expertise that the trait is the way to go!